src/module.cc

```diff
@@ -0,0 +1,338 @@
+  ProcessNodeReportFileName(*parameter);
+}
+NAN_METHOD(SetDirectory) {
+  Nan::Utf8String parameter(info[0]->ToString());
+  ProcessNodeReportDirectory(*parameter);
+}
+NAN_METHOD(SetVerbose) {
+  Nan::Utf8String parameter(info[0]->ToString());
+  nodereport_verbose = ProcessNodeReportVerboseSwitch(*parameter);
+}
+
+/*******************************************************************************
+ * Callbacks for triggering NodeReport on failure events (as configured)
+ *  - fatal error
+ *  - uncaught exception
+ *  - signal
+ ******************************************************************************/
+static void OnFatalError(const char* location, const char* message) {
+  if (location) {
+    fprintf(stderr, "FATAL ERROR: %s %s\n", location, message);
+  } else {
+    fprintf(stderr, "FATAL ERROR: %s\n", message);
+  }
+  // Trigger NodeReport if requested
+  if (nodereport_events & NR_FATALERROR) {
+    TriggerNodeReport(Isolate::GetCurrent(), kFatalError, message, location, NULL);
+  }
+  fflush(stderr);
+  if (nodereport_core) {
+    raise(SIGABRT); // core dump requested
+  } else {
+    exit(0); // no core dump requested
+  }
+}
+
+bool OnUncaughtException(v8::Isolate* isolate) {
+   // Trigger NodeReport if required
+  if (nodereport_events & NR_EXCEPTION) {
+    TriggerNodeReport(Isolate::GetCurrent(), kException, "exception", "OnUncaughtException (nodereport/src/module.cc)", NULL);
+  } 
+  if (nodereport_core) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+#ifndef _WIN32
+static void SignalDumpInterruptCallback(Isolate *isolate, void *data) {
+  if (report_signal != 0) {
+    fprintf(stderr,"SignalDumpInterruptCallback - handling signal\n");
+    if (nodereport_events & NR_SIGNAL) {
+      if (nodereport_verbose) {
+        fprintf(stderr,"SignalDumpInterruptCallback - triggering NodeReport\n");
+      }
+      TriggerNodeReport(Isolate::GetCurrent(), kSignal_JS,
+                        node::signo_string(*(static_cast<int *>(data))),
+                        "node::SignalDumpInterruptCallback()", NULL);
+    }
+    report_signal = 0;
+  }
+}
+static void SignalDumpAsyncCallback(uv_async_t* handle) {
+  if (report_signal != 0) {
+    fprintf(stderr,"SignalDumpAsyncCallback - handling signal\n");
+    if (nodereport_events & NR_SIGNAL) {
+      if (nodereport_verbose) {
+        fprintf(stderr,"SignalDumpAsyncCallback - triggering NodeReport\n");
+      }
+      size_t signo_data = reinterpret_cast<size_t>(handle->data);
+      TriggerNodeReport(Isolate::GetCurrent(), kSignal_UV,
+                        node::signo_string(static_cast<int>(signo_data)),
+                        "node::SignalDumpAsyncCallback()", NULL);
+    }
+    report_signal = 0;
+  }
+}
+
+static void RegisterSignalHandler(int signal, void (*handler)(int signal), bool reset_handler = false) {
+  struct sigaction sa;
+  memset(&sa, 0, sizeof(sa));
+  sa.sa_handler = handler;
+  sa.sa_flags = reset_handler ? SA_RESETHAND : 0;

```

Comment:
SA_RESETHAND doesn't work with some versions of freebsd's libthr.  The workaround is to set the handler to SIG_DFL in the signal handler.